File: Lab2/c/main.c
#include <stdio.h>
#include <string.h>
#include "task1.c"
#include "task2.c"
#include "task3.c"
#include "validations.c"

int main(int argc, char *argv[]) {
	if (argc != 2) {
		printf("Se debe usar: %s <cadena>\n", argv[0]);
		return 1;
	}

    if (strlen(argv[1]) > 100) {
		printf("Error: la cadena supera los 100 caracteres.\n");
		return 1;
	}

    if (!clean_validate(argv[1])) {
        printf("Error: la cadena no contiene caracteres válidos.\n");
        return 1;
    }
	
    reverse_string(argv[1]);
    int count[7] = {0};
    count_vowels_consonants(argv[1], count);
    printf("%s %d ", argv[1], count[0]);
    for (int i = 1; i <= 5; i++) {
        if (count[i] > 0) {
            printf("%d ", count[i]);
        }
    }
    printf("%d ", count[6]);
    change_spaces_to_underscore(argv[1]);
    printf("%s\n", argv[1]);
	return 0;
}

File: Lab2/go/main.go
package main

import (
	"fmt"
	"os"
)

func main() {
    if len(os.Args) != 2 {
        fmt.Printf("Se debe usar: %s <cadena>\n", os.Args[0])
        os.Exit(1)
    }

    data := []byte(os.Args[1])

    if len(data) > 100 {
        fmt.Println("Error: la cadena supera los 100 caracteres.")
        os.Exit(1)
    }
    
    if validCount := cleanValidate(data); validCount == 0 {
        fmt.Println("Error: la cadena no contiene caracteres válidos.")
        os.Exit(1)
    }
    
    reverseString(data)

    counts := make([]int, 7)
    countVowelsConsonants(data, counts)

    
    fmt.Printf("%s ", string(data))

    for i := 0; i < 6; i++ {
        if i == 0 || counts[i] > 0 { 
             fmt.Printf("%d ", counts[i])
        }
    }
    fmt.Printf("%d ", counts[6])


    changeSpacesToUnderscore(data)

    fmt.Printf("%s\n", string(data))
}